File: dodam-application/dodam-rest-api/src/main/java/b1nd/dodam/restapi/notice/presentation/NoticeController.java
package b1nd.dodam.restapi.notice.presentation;

import b1nd.dodam.restapi.notice.application.NoticeUseCase;
import b1nd.dodam.restapi.notice.application.data.req.GenerateNoticeReq;
import b1nd.dodam.restapi.notice.application.data.res.NoticeRes;
import b1nd.dodam.restapi.support.data.Response;
import b1nd.dodam.restapi.support.data.ResponseData;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/notice")
@RequiredArgsConstructor
public class NoticeController {

    private final NoticeUseCase noticeUseCase;

    @PostMapping
    public ResponseData<Long> generate(@RequestBody GenerateNoticeReq generateNoticeReq){
        return noticeUseCase.register(generateNoticeReq);
    }

    @GetMapping
    public ResponseData<List<NoticeRes>> getByStatus(
            @RequestParam String keyword,
            @RequestParam(required = false) Long lastId,
            @RequestParam int limit
    ){
        return noticeUseCase.getNotices(keyword, lastId, limit);
    }

    @GetMapping("/division")
    public ResponseData<List<NoticeRes>> getById(
            @RequestParam(required = false) Long id,
            @RequestParam(required = false) Long lastId,
            @RequestParam int limit
    ){
        return noticeUseCase.getNoticesByDivision(id, lastId, limit);
    }

    @DeleteMapping("/{id}")
    public Response deleteById(@PathVariable Long id){
        return noticeUseCase.deleteNotice(id);
    }

}

File: services/service-notice/src/main/kotlin/com/b1nd/dodamdodam/notice/presentation/notice/NoticeController.kt
package com.b1nd.dodamdodam.notice.presentation.notice

import com.b1nd.dodamdodam.core.common.data.Response
import com.b1nd.dodamdodam.notice.application.notice.NoticeUseCase
import com.b1nd.dodamdodam.notice.application.notice.data.request.GenerateNoticeRequest
import com.b1nd.dodamdodam.notice.application.notice.data.request.ModifyNoticeRequest
import com.b1nd.dodamdodam.notice.application.notice.data.response.NoticeResponse
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/notice")
class NoticeController(
    private val noticeUseCase: NoticeUseCase
) {
    @PostMapping
    fun generate(@RequestBody request: GenerateNoticeRequest): Response<Long> {
        return noticeUseCase.register(request)
    }

    @GetMapping
    fun getNotices(
        @RequestParam(required = false) keyword: String?,
        @RequestParam(required = false) lastId: Long?,
        @RequestParam(defaultValue = "10") limit: Int
    ): Response<List<NoticeResponse>> {
        return noticeUseCase.getNotices(keyword, lastId, limit)
    }

    @PatchMapping("/{id}")
    fun modify(
        @PathVariable id: Long,
        @RequestBody request: ModifyNoticeRequest
    ): Response<Unit> {
        return noticeUseCase.modify(id, request)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): Response<Unit> {
        return noticeUseCase.delete(id)
    }

    @GetMapping("/health")
    fun health(): String = "OK"
}
